src/out/transformer/distributed_tensor/softmax.check.cu

```diff
@@ -0,0 +1,191 @@
+    // begin creating and setting tensor descriptor of shape List(2, 1, 3, 3)
+    cudnnTensorDescriptor_t x39;
+    CUDNNCHECK(cudnnCreateTensorDescriptor(&x39));
+    CUDNNCHECK(cudnnSetTensor4dDescriptor(x39, CUDNN_TENSOR_NCHW, CUDNN_DATA_FLOAT, 2, 1, 3, 3));
+    // end creating and setting tensor descriptor
+    // begin allocating gpu array for the gradient of input of softmax
+    CUDA_CALL(cudaSetDevice(x7));
+    float* x40 = (float*)malloc(0 * sizeof(float));
+    CUDA_CALL(cudaMalloc(&x40, (size_t)(18 * sizeof(float))));
+    // end allocating gpu array for the gradient of input of softmax
+    // begin softmax backward pass
+    float x41 = 1.0;
+    float x42 = 0.0;
+    CUDNNCHECK(cudnnSoftmaxBackward(x8, CUDNN_SOFTMAX_FAST, CUDNN_SOFTMAX_MODE_INSTANCE, &x41, &x39, &x11, x39, &x37, &x42, x39, &x40));
+    // end softmax backward pass
+    // begin computing ACCUM on GPU for size 9 and type Float at device (pre-rename) x39 with base_operand x86 and addition_operand x214
+    CUDA_CALL(cudaSetDevice(x7));
+    x21<<<dim3(28, 1, 1), dim3(512, 1, 1)>>>(x12, x40, 9);
+    // end computing ACCUM on GPU for size 9 and type Float at device (pre-rename) x39 with base_operand x86 and addition_operand x214
+    // begin computing SGD on GPU for size 9 and type Float at device (pre-name) x39 with weight x70, grad x86, and momentum x124
+    CUDA_CALL(cudaSetDevice(x7));
+    x28<<<dim3(28, 1, 1), dim3(512, 1, 1)>>>(x11, x12, x19, 9);
+    // end computing SGD on GPU for size 9 and type Float at device (pre-name) x39 with weight x70, grad x86, and momentum x124
+    x20 = x20 + 1;
+  }
+  // Only declare recv buffer if this is the root
+  bool x43 = x7 == 0;
+  float* x44 = x43 ? ({
+    float* x45 = (float*)malloc(0 * sizeof(float));
+    CUDA_CALL(cudaMalloc(&x45, (size_t)(18 * sizeof(float))));
+    x45;
+  }) : ({
+    float* x46 = (float*)malloc(0 * sizeof(float));
+    CUDA_CALL(cudaMalloc(&x46, (size_t)0));
+    x46;
+  });
+  // Gather by groups of NCCL send/recv
+  NCCLCHECK(ncclGroupStart());
+  ncclResult_t x47 = ncclSend(x11, (size_t)18, ncclFloat32, 0, x5, x6);
+  NCCLCHECK(x47);
+  if (x43) {
+    int x48 = x1;
+    int x49 = 0;
+    while (x49 != x48) {
+      int x50 = x49;
+      NCCLCHECK(ncclRecv(x44 + x50 * 9, (size_t)18, ncclFloat32, x50, x5, x6));
+      x49 = x49 + 1;
+    }
+  }
+  NCCLCHECK(ncclGroupEnd());
+  // print the array only if this is the root
+  if (x43) {
+    // begin copying GPU array x298 to CPU and print for size 18 and type Float
+    float* x51 = (float*)malloc(18 * sizeof(float));
+    CUDA_CALL(cudaMemcpy(x51, x44, (size_t)(18 * sizeof(float)), cudaMemcpyDeviceToHost));
+    int x52 = 0;
+    while (x52 != 18) {
+      printf("%f ", x51[x52]);
+      x52 = x52 + 1;
+    }
+    printf("\n");
+    // end copying GPU array x298 to CPU and print for size 18 and type Float
+  }
+  printf("compile");
+  CUDNNCHECK(cudnnDestroy(x8));
```

Comment:
should also include CUDNNCHECK(cudnnDestroyTensorDescriptor(x39)), since x39 is a descriptor created, and should be destroyed by finalize_cudnn